lib/Storage.js

```diff
@@ -1,87 +1,132 @@
   }
 
   const storageInterface = {
+    /**
+     * Sets an item in the prefixed storage.
+     * @returns {Promise}
+     * @resolves With the stored value, or null.
+     * @rejects Javascript exception.
+     */
     getItem(keySuffix) {
-      return idbPromise.then(db => {
-        return new sandbox.Promise((resolve, reject) => {
-          let key = makeKey(keySuffix);
-          let transaction = db.transaction(['storage'], 'readonly');
-          let request = transaction.objectStore('storage').get(key);
-          request.onsuccess = event => {
-            if (event.target.result) {
-              resolve(event.target.result.value);
-            } else {
-              resolve(null);
-            }
-          };
-          request.onerror = err => reject(err);
+      return loadStorage()
+        .then(store => {
+          let namespace = store.data[prefix] || {};
+          const value = namespace[keySuffix] || null;
+          return sandbox.Promise.resolve(Cu.cloneInto(value, sandbox));
+        })
+        .catch(err => {
+          Log.error(err);
+          throw new Error(err.toString());
         });
-      });
     },
 
+    /**
+     * Sets an item in the prefixed storage.
+     * @returns {Promise}
+     * @resolves When the operation is completed succesfully
+     * @rejects Javascript exception.
+     */
     setItem(keySuffix, value) {
-      return idbPromise.then(db => {
-        return new sandbox.Promise((resolve, reject) => {
-          let key = makeKey(keySuffix);
-          let transaction = db.transaction(['storage'], 'readwrite');
-          let request = transaction.objectStore('storage').put({key, value});
-          request.onsuccess = () => resolve();
-          request.onerror = err => reject(err);
+      return loadStorage()
+        .then(store => {
+          if (!(prefix in store.data)) {
+            store.data[prefix] = {};
+          }
+          store.data[prefix][keySuffix] = value;
+          return store.save();
+        })
+        .then(() => {
+          return sandbox.Promise.resolve();
+        })
+        .catch(err => {
+          Log.error(err);
+          throw new Error(err.toString());
         });
-      });
     },
 
+    /**
+     * Removes a single item from the prefixed storage.
+     * @returns {Promise}
+     * @resolves When the operation is completed succesfully
+     * @rejects Javascript exception.
+     */
     removeItem(keySuffix) {
-      return idbPromise.then(db => {
-        return new sandbox.Promise((resolve, reject) => {
-          let key = makeKey(keySuffix);
-          let transaction = db.transaction(['storage'], 'readwrite');
-          let request = transaction.objectStore('storage').delete(key);
-          request.onsuccess = () => resolve();
-          request.onerror = err => reject(err);
+      return loadStorage()
+        .then(store => {
+          if (!(prefix in store.data)) {
+            return Promise.resolve();
```

Comment:
Return values are already wrapped in a promise, so you can just return `true` or whatever here.